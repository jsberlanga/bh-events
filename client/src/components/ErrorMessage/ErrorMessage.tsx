import * as React from 'react';
import styled from 'styled-components';

const StyledMessage = styled.div`
  display: block;
  background: #af2d2d;
  color: #fff;
  padding: 10px;
`;

const ErrorMessage: React.FC = ({ children }) => (
  <StyledMessage role="alert">{children}</StyledMessage>
);

export default ErrorMessage;
