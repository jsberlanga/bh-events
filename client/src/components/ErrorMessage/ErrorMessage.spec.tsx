import { render, screen } from '@testing-library/react';
import ErrorMessage from './ErrorMessage';

describe('CreateEvent', () => {
  it('should render with styles correctly', () => {
    const { container } = render(<ErrorMessage>An error ocurred</ErrorMessage>);

    expect(container).toMatchSnapshot();
  });

  it('should have the "alert" role', () => {
    render(<ErrorMessage>An error ocurred</ErrorMessage>);

    expect(screen.getByRole('alert')).toBeInTheDocument();
  });
});
