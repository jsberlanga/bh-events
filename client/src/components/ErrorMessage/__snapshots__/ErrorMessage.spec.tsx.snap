// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateEvent should render with styles correctly 1`] = `
.c0 {
  display: block;
  background: #af2d2d;
  color: #fff;
  padding: 10px;
}

<div>
  <div
    class="c0"
    role="alert"
  >
    An error ocurred
  </div>
</div>
`;
